sdk/core/azure-core/src/main/java/com/azure/core/http/policy/RequestIdPolicyOptions.java

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.http.policy;
+
+import com.azure.core.annotation.Immutable;
+import com.azure.core.http.HttpHeaders;
+
+import java.util.Objects;
+import java.util.function.Supplier;
+
+/**
+ * Immutable Configuration options for {@link RequestIdPolicy}.
+ * User can provide a {@link Supplier} to generated id for various id headers to be used in
+ * {@link com.azure.core.http.HttpRequest}. Examples of such headers are 'x-ms-client-request-id
+ * or 'x-ms-correlation-request-id' etc.
+ */
+@Immutable
+public class RequestIdPolicyOptions {
```

Comment:
Do we need a separate type for this? Can the `Supplier` instead be passed to `RequestIdPolicy` constructor directly?

Reply:
Agreed on this comment, I don't think this policy will require additional configuration outside of being able to set Request ID headers.

Reply:
agree